src/sample_with_replacement.js

```diff
@@ -0,0 +1,41 @@
+'use strict';
+/* @flow */
+
+/**
+ * Sampling with replacement is a type of sampling that allows the same
+ * item to be picked out of a population more than once.
+ *
+ * @param {Array} population an array of any kind of element
+ * @param {number} n count of how many elements to take
+ * @param {Function} [randomSource=Math.random] an optional entropy source
+ * @return {Array} n sampled items from the population
+ * @example
+ * var sample = sampleWithReplacement([1, 2, 3, 4], 2);
+ * sampleWithReplacement; // = [2, 4] or any other random permutation
```

Comment:
I think `permutation` implies sampling _without_ replacement, since a permutation is technically a rearrangement (so it couldn't contain duplicates).